src/deepsparse/server/openai_server.py

```diff
@@ -0,0 +1,385 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import time
+from functools import partial
+from http import HTTPStatus
+from typing import AsyncGenerator, Dict, List, Optional
+
+from deepsparse import Pipeline
+from deepsparse.server.config import EndpointConfig
+from deepsparse.server.helpers import create_error_response
+from deepsparse.server.output import CompletionOutput, RequestOutput
+from deepsparse.server.protocol import (
+    ChatCompletionRequest,
+    ChatCompletionResponse,
+    ChatCompletionResponseChoice,
+    ChatCompletionResponseStreamChoice,
+    ChatCompletionStreamResponse,
+    ChatMessage,
+    DeltaMessage,
+    ModelCard,
+    ModelList,
+    ModelPermission,
+    UsageInfo,
+    random_uuid,
+)
+from deepsparse.server.server import ProxyPipeline, Server
+from fastapi import BackgroundTasks, FastAPI, Request
+from fastapi.responses import StreamingResponse
+
+
+_LOGGER = logging.getLogger(__name__)
+
+OPENAI_CHAT_NOT_SUPPORTED = ["logit_bias", "best_ok", "ignore_eos", "use_beam_search"]
+OPENAI_TO_DEEPSPARSE_MAPPINGS = {
+    "max_tokens": "max_length",
+    "frequency_penalty": "repetition_penalty",
+}
+SUPPORTED_TASKS = ["text_generation", "opt", "bloom"]
```

Comment:
should use `SupportedTasks.is_X` to standardize on task aliases as well as str format https://github.com/neuralmagic/deepsparse/blob/b71ebffa71e653746b6b022ca7bd4788d2252c6c/src/deepsparse/tasks.py#L76